server/src/internalClusterTest/java/org/opensearch/indices/replication/SearchReplicaRestoreIT.java

```diff
@@ -0,0 +1,283 @@
+        assertHitCount(resp, DOC_COUNT);
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnSegRepWithSearchReplica_RestoreOnDocRep() throws Exception {
+        bootstrapIndexWithSearchNodes(ReplicationType.SEGMENT);
+        createRepoAndSnapshot(REPOSITORY_NAME, FS_REPOSITORY_TYPE, SNAPSHOT_NAME, INDEX_NAME);
+
+        SnapshotRestoreException exception = expectThrows(SnapshotRestoreException.class,
+            () ->  restoreSnapshot(REPOSITORY_NAME, SNAPSHOT_NAME, INDEX_NAME, RESTORED_INDEX_NAME,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT)
+                    .build()
+            ));
+        assertTrue(exception.getMessage().contains(getSnapshotExceptionMessage(ReplicationType.SEGMENT, ReplicationType.DOCUMENT)));
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnSegRepWithSearchReplica_RestoreOnDocRepWithNoSearchReplica() throws Exception {
+        bootstrapIndexWithSearchNodes(ReplicationType.SEGMENT);
+        createRepoAndSnapshot(REPOSITORY_NAME, FS_REPOSITORY_TYPE, SNAPSHOT_NAME, INDEX_NAME);
+
+        restoreSnapshot(REPOSITORY_NAME, SNAPSHOT_NAME, INDEX_NAME, RESTORED_INDEX_NAME,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT)
+                .put(IndexMetadata.SETTING_NUMBER_OF_SEARCH_REPLICAS, 0)
+                .build()
+        );
+        ensureGreen(RESTORED_INDEX_NAME);
+
+        SearchResponse resp = client().prepareSearch(RESTORED_INDEX_NAME).setQuery(QueryBuilders.matchAllQuery()).get();
+        assertHitCount(resp, DOC_COUNT);
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnSegRepWithSearchReplica_RestoreOnSegRep() throws Exception {
+        bootstrapIndexWithSearchNodes(ReplicationType.SEGMENT);
+        createRepoAndSnapshot(REPOSITORY_NAME, FS_REPOSITORY_TYPE, SNAPSHOT_NAME, INDEX_NAME);
+
+        restoreSnapshot(REPOSITORY_NAME, SNAPSHOT_NAME, INDEX_NAME, RESTORED_INDEX_NAME,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
+                .build()
+        );
+        ensureGreen(RESTORED_INDEX_NAME);
+
+        SearchResponse resp = client().prepareSearch(RESTORED_INDEX_NAME).setQuery(QueryBuilders.matchAllQuery()).get();
+        assertHitCount(resp, DOC_COUNT);
+    }
+
+    private void bootstrapIndexWithOutSearchNodes(ReplicationType replicationType) throws InterruptedException {
+        startCluster(2);
+        createIndex(INDEX_NAME, getIndexSettings(1, 1, 0,
+            replicationType));
+        indexRandomDocs(INDEX_NAME, DOC_COUNT);
+        refresh(INDEX_NAME);
+        ensureGreen(INDEX_NAME);
+    }
+
+    private void bootstrapIndexWithSearchNodes(ReplicationType replicationType) throws InterruptedException {
```

Comment:
you can't boostrap with docrep here, so I don't think you need the param?

Reply:
Thats right